### PR TITLE
updating Upload request serializers with two additional fields #1866

### PR DESCRIPTION
**Pull request by Naruto0**
Signed-off-by: Filip Hron <filip.hron@gmail.com>

### CODE REVIEW

File: weblate/api/views.py [hunk 1]
```diff
@@ -454,6 +454,8 @@ def file(self, request, **kwargs):
             request,
             serializer.validated_data['file'],
             serializer.validated_data['overwrite'],
+            serializer.validated_data['email'],
+            serializer.validated_data['author'],
```
Comment (nijel) [inline weblate/api/views.py]: What it should additionally do here is to check permissions - not all users are allowed to set email and author, see related code in the UI:

https://github.com/WeblateOrg/weblate/blob/8b43331590c33ec19a14e6f214e103db94f8dfb3/weblate/trans/forms.py#L634

Comment (Naruto0) [reply weblate/api/views.py]: Thank you for an update. I will try to implement also.

File: weblate/api/views.py [hunk 2]
```diff
@@ -450,11 +450,15 @@ def file(self, request, **kwargs):
...
+        can_author = request.user.has_perm('upload.authorship', obj)
+
         not_found, skipped, accepted, total = obj.merge_upload(
             request,
             serializer.validated_data['file'],
             serializer.validated_data['overwrite'],
-        )
+            serializer.validated_data['email'] if can_author else None,
+            serializer.validated_data['author'] if can_author else None,
+            )
```
[the remark this answers is not in the document]
Comment (nijel) [reply weblate/api/views.py]: @Naruto0 can you please fix this?

[the remark this answers is not in the document]
Comment (Naruto0) [reply weblate/api/views.py]: @nijel Hi, yes.

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**